pandas/core/ops/__init__.py

```diff
@@ -748,7 +748,7 @@ def f(self, other, axis=default_axis, level=None):
             return _combine_series_frame(self, other, op, axis=axis)
         else:
             # in this case we always have `np.ndim(other) == 0`
```

Comment:
see my comment on your other PR. I would rather not do this as another arg, better is to create a NamedTuple that holds both and pass that around

Reply:
another alternative would be to make the getting of str_rep a dict lookup in expressions